File: app/src/main/java/com/master/kit/TestActivity/CameraActivity.java
package com.master.kit.TestActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;

import com.master.kit.Base.BaseActivity;
import com.master.kit.R;
import com.master.kit.utils.AlbumUtil;
import com.master.kit.utils.LogUtil;

public class CameraActivity extends BaseActivity {

    private ImageView imageView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_camera);
        findViewById(R.id.btn_main).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
               // CameraUtil.takePhoto(CameraActivity.this, Environment.getExternalStorageDirectory().getPath(),"aa",1000);
                AlbumUtil.selectPhoto(CameraActivity.this,100);
            }
        });
        imageView = (ImageView) findViewById(R.id.iv_main);

    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        LogUtil.e("Intent",data+"");
        if(resultCode == 0) return;
        imageView.setImageBitmap(AlbumUtil.getBitmapFromIntent(this,data));
    }

}

File: other/src/main/java/com/ooftf/master/other/activity/CameraActivity.kt
package com.ooftf.master.other.activity

import android.Manifest
import android.app.Activity
import android.content.Intent
import android.graphics.BitmapFactory
import android.os.Bundle
import android.util.Log
import com.alibaba.android.arouter.facade.annotation.Route
import com.ooftf.master.other.R
import com.ooftf.master.other.engine.HiCamera
import com.ooftf.service.base.BaseActivity
import com.ooftf.service.constant.ProviderConstant
import com.tbruyelle.rxpermissions2.RxPermissions
import kotlinx.android.synthetic.main.activity_camera.*


@Route(path = "/other/activity/camera")
class CameraActivity : BaseActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_camera)
        button.setOnClickListener {
            checkPermissionsToCamera();
        }
    }

    private fun checkPermissionsToCamera() {
        RxPermissions(this)
                .request(Manifest.permission.CAMERA)
                .subscribe {
                    if (it) {
                        camera()
                    } else {
                        toast("获取相机权限失败")
                    }
                }
    }

    var hiCamera: HiCamera? = null
    private fun camera() {
        hiCamera = HiCamera
                .from(this)
                .setFileProviderAuthority(ProviderConstant.FILE_PROVIDER)
                .forResult(REQUEST_CODE_CAMERA)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == REQUEST_CODE_CAMERA && resultCode == Activity.RESULT_OK) {
            //操作File对象
            Log.e("hiCamera", hiCamera?.resultFile!!.path)
            imageView.setImageBitmap(BitmapFactory.decodeFile(hiCamera!!.resultFile!!.absolutePath))
        }
    }

    companion object {
        val REQUEST_CODE_CAMERA = 456
    }
}